### ISSUE TITLE
Accessibility: Custom markers should support 'alt' attribute

### ISSUE THREAD

**Issue by rhewitt22** [Leaflet/Leaflet#4798]
There is no way to supply alt text when creating custom map markers.

Alt text is read aloud to those using a screen reader and is required for all images under section 508.

**Comment by IvanSanchez** [comment]
Yes, there is a way. http://leafletjs.com/reference-1.0.0.html#marker-alt

**Comment by rhewitt22** [comment]
My mistake I was looking at L.Icon.

**Comment by andymantell** [comment]
Worth noting that if you set this to an empty string, leaflet doesn't display the alt text, which gets flagged up as an accessiblilty failure. I'm not aware of any use case for this which would require _no_ alt text at all. Empty strings are used to denote a decorational image so ideally Leaflet would always render the alt text rather than doing this in Marker.js:

```
if (options.alt) {
    icon.alt = options.alt;
}
```

Because empty string casts to boolean false, the alt text never gets set. A temporary workaround when creating the marker is to pass in the following object as alt text:

```
var altText = {}
altText.toString = function () {
      return ''
}

altText.valueOf = function () {
    return true
}

L.marker( ... , { alt: altText })
```

This fools leaflet into displaying the alt text, because the `valueOf` method returns `true`. But then when the alt text is rendered, the `toString` method returns an empty string.

Ideally the `if(options.alt)` line in Marker.js would be removed so that this little hack isn't necessary, but for anyone struggling with this in the meantime, this should help.

**Comment by cherniavskii** [comment]
@andymantell if it's a bug - please open a new issue. Adding a comment to closed unrelated issue is not a but report ;)

**Comment by andymantell** [comment]
Yes good point sorry, will raise one now!

**Comment by andymantell** [comment]
See #5953
